sysdig/resource_sysdig_secure_cloud_auth_account.go

```diff
@@ -0,0 +1,174 @@
+package sysdig
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	v2 "github.com/draios/terraform-provider-sysdig/sysdig/internal/client/v2"
+	cloudauth "github.com/draios/terraform-provider-sysdig/sysdig/internal/client/v2/cloudauth/go"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceSysdigSecureCloudauthAccount() *schema.Resource {
+	timeout := 5 * time.Minute
+
+	return &schema.Resource{
+		CreateContext: resourceSysdigSecureCloudauthAccountCreate,
+		UpdateContext: resourceSysdigSecureCloudauthAccountUpdate,
+		ReadContext:   resourceSysdigSecureCloudauthAccountRead,
+		DeleteContext: resourceSysdigSecureCloudauthAccountDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(timeout),
+			Update: schema.DefaultTimeout(timeout),
+			Read:   schema.DefaultTimeout(timeout),
+			Delete: schema.DefaultTimeout(timeout),
+		},
+		Schema: map[string]*schema.Schema{
+			"account_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"cloud_provider_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"cloud_provider_type": {
+				Type:         schema.TypeInt,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{cloudauth.Provider_PROVIDER_AWS.String(), cloudauth.Provider_PROVIDER_GCP.String(), cloudauth.Provider_PROVIDER_AZURE.String()}, false),
+			},
+			"enabled": {
+				Type:     schema.TypeBool,
+				Required: true,
+			},
+		},
+	}
+}
+
+func getSecureCloudauthAccountClient(client SysdigClients) (v2.CloudauthAccountSecureInterface, error) {
+	return client.sysdigSecureClientV2()
+}
+
+func resourceSysdigSecureCloudauthAccountCreate(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, err := getSecureCloudauthAccountClient((meta.(SysdigClients)))
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	cloudauthAccount, err := client.CreateCloudauthAccountSecure(ctx, cloudauthAccountFromResourceData(data))
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	data.SetId(cloudauthAccount.Id)
+
+	return nil
+}
+
+func resourceSysdigSecureCloudauthAccountRead(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, err := getSecureCloudauthAccountClient(meta.(SysdigClients))
+
+	if err != nil {
+		data.SetId("")
+		return diag.FromErr(err)
+	}
+
+	cloudauthAccount, err := client.GetCloudauthAccountSecure(ctx, data.Id())
+
+	if err != nil {
+		data.SetId("")
```

Comment:
Why are we changing data here? Is it necessary?

Reply:
I don't think it's necessary. I was following the standard we have for the existing cloud account resource.